$NetBSD$

1) Set the RPATH when creating executables and shared libraries.
2) Use pkgsrc libtommath instead of cyclone libtommath (cyclonebn).
3) Fix the tests for open_memstream and fmemopen.

--- Makefile.config.orig	2024-02-14 02:29:41.000000000 +0000
+++ Makefile.config
@@ -21,8 +21,8 @@ CYC_PTHREAD_SET_STACK_SIZE ?=
 OS = $(shell uname)
 CC ?= cc
 
-LIBS = -pthread -lcyclone -lck -lm -lcyclonebn
-ifneq ($(OS),FreeBSD)
+LIBS = -pthread -lcyclone -lck -lm -ltommath
+ifeq (,$(findstring $(OS),FreeBSD NetBSD OpenBSD DragonFly))
 # libdl is part of libc on FreeBSD
 LIBS += -ldl
 endif
@@ -55,6 +55,12 @@ CREATE_LIBRARY_COMMAND = $(AR)
 CREATE_LIBRARY_FLAGS = rcs
 endif
 
+COMP_LIBDIRS+= -L$(PKGSRC_LIBDIR)
+COMP_LIBDIRS+= -L$(PKGSRC_LIBDIR)/cyclone-bootstrap
+COMP_LIBDIRS+= $(COMPILER_RPATH_FLAG),$(PKGSRC_LIBDIR)
+LDFLAGS     += -L$(PKGSRC_LIBDIR)
+LDFLAGS     += $(COMPILER_RPATH_FLAG),$(PKGSRC_LIBDIR)
+
 # /usr/local is not in the search path by default on FreeBSD, so if libtommath and/or 
 # concurrencykit was installed via Ports, it won't be picked up without explicitly looking
 # for it here
@@ -91,8 +97,8 @@ DESTDIR   ?=
 
 # Automatically detect platform-specific flags, instead of using autoconf
 #CYC_PLATFORM_HAS_MEMSTREAM ?= 1
-CYC_PLATFORM_HAS_MEMSTREAM := $(shell echo "main(){char *buf; int len; open_memstream(&buf, &len);}" | $(CC) -xc - >/dev/null 2>/dev/null && echo 1 || echo 0)
-CYC_PLATFORM_HAS_FMEMOPEN := $(shell echo "main(){char *buf; fmemopen(&buf, 0, \"r\");}" | $(CC) -xc - >/dev/null 2>/dev/null && echo 1 || echo 0)
+CYC_PLATFORM_HAS_MEMSTREAM := $(shell echo "int main(){char *buf; int len; open_memstream(&buf, &len);}" | $(CC) -xc -include stdio.h - >/dev/null 2>/dev/null && echo 1 || echo 0)
+CYC_PLATFORM_HAS_FMEMOPEN := $(shell echo "int main(){char *buf; fmemopen(&buf, 0, \"r\");}" | $(CC) -xc -include stdio.h - >/dev/null 2>/dev/null && echo 1 || echo 0)
 
 # code from chibi's makefile to detect platform
 ifndef PLATFORM
